java-checks/src/main/java/org/sonar/java/checks/ThrowCheckedExceptionCheck.java

```diff
@@ -41,18 +46,53 @@
 @SqaleConstantRemediation("1h")
 public class ThrowCheckedExceptionCheck extends SubscriptionBaseVisitor {
 
+  private Stack<MethodTree> methods = new Stack<>();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace the synchronized class "Stack" by an unsynchronized one such as "Deque". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1149)